wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Replace `findViewById` with `ViewBinding` for Site Picker (`relatively safe`)

Parent #18906

This PR's main focus is to replace `findViewById ` with `ViewBinding` for the [SitePickerActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/5b0d3f39c8cfe9c08b4e1547e46bc947f9be6b8f/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java#L86) class.

Additionally, this PR is also dealing with adding any missing nullability annotations to this activity class, focusing mainly on its fields, and, any effected neighbour classes. But first, all existing warnings on this class are being resolved/suppressed so as to make it easier to deal with any missing nullability checks (with help from the IDE), especially when a `@Nullable` annotation is added.

FYI: This change is `relatively safe`, meaning that although there are compile-time changes associated with this change, and it needs testing, the changes included in this PR are very targeted and specific to one screen, and one screen only. Thus, if this screen is tested appropriately, it should be safe to merge this to `trunk`.

-----

PS: @thomashorta I added you as the main reviewer, randomly so, since I just wanted someone from the Jetpack/WordPress mobile team to be aware of and sign-off on that change for JP/WPAndroid. Feel free to merge this PR directly yourself if you deem so.

-----

`findViewById ` -> `ViewBinding`

1. [Replace findviewbyid with viewbinding for site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/777de0b13c2ebe07f2be8e27ae947bc0116f895e)
2. [Use non-null viewbinding method parameter for site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/f1d3930921a951c1109b87118eb515429485dd7a)

Nullability Annotation List:

1. [Add missing nullability annotations to sdk override methods](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/7aae6bf080716f9ad76d91c7685a3f016e35ec90)
2. [Add missing nl-a to adapter field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/6f90b037f950d09900c6a6bd9390f4d6fab037ee)
3. [Add missing nl-a to str helper field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/5c869868c28b7fc3e84068a2969f91569d90b404)
4. [Add missing nl-a to action mode field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/7b62a772996eb0bda6847fb9f2f406fbba40c22f)
5. [Add missing nl-a to reblog action mode field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/fa981265da2afa9c9b1efaba4348f5629163aba9)
6. [Add missing nl-a to menu edit field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/2e589b703d15447b4e122b230778bbd9d8044e24)
7. [Add missing nl-a to menu add field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/bc370ad8a57eb8c75e4bac8ff5d981cd4dd053a0)
8. [Add missing nl-a to menu search field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/782c843808dd63444a5ffa2cf41e72c510339af6)
9. [Add missing nl-a to search view field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/f37ad70f66832e7db3e01a6d271978b3891d710f)
10. [Add missing nl-a to site picker mode field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/07fb5a51773c64f9347b2363e07150acfcd027e7)
11. [Add missing nn-a to debouncer field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/4d28cdd97247e882777fef23b58daf58a539b75d)
12. [Add missing nl-a to view model field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/0cee0cf27353b9ecf31468d80bbbc1fe71792c64)
13. [Add missing nn-a to selected positions field](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/7b792f30c15c749fa65ca15061fc996560a55c3f)
14. [Add missing nl-a to private restore sis method parameter](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/0431fe300d92ef02b18bc94ae7e93a7810053190)

Warnings Resolution List:

1. [Guard nullable site to reblog variable on site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/6886c1678fe63c7e32ac46645a8542ded90b190e)
2. [Replace unnecessary switch with if statement on site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/bdf37408ee0be432bb60fc0d9f75b6754e903263)
3. [Guard nullable site model variable on site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/51dff5467ff0c9c43f7486d25290ca1f8f66cb47)

Warnings Suppression List:

1. [Suppress unchecked warning on site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/fafdf4bdcb142ef65a6731c00853353d0409d8eb)
2. [Suppress deprecation warnings on site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/321eb18965c89143f7916e38ee5097307948b299)

Cleanup List:

1. [Remove unnecessary feature config injected field on site picker](https://github.com/wordpress-mobile/WordPress-Android/pull/19209/commits/82e169ee635cc63e321c27d501235a4d7bbc2689)

-----

## To test:

1. Quickly smoke test any site picker related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected.
16. In addition to the above smoke test, you can expand the below and follow the inner and more explicit test steps within:

<details>
    <summary>Site Picker Screen [SitePickerActivity.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `My Site` tab and tap on the `down-arrow`(`top-right`).
- Verify that the `Site Picker` screen is shown and functioning as expected.
- For example, try switching sites, searching for a site, adding a new site and showing/hiding sites.

</details>

FYI: While I was testing the `Site Picker` screen I noticed a bug. This bug is not caused by any of the changes included in this PR, and, can be easily reproduced on `trunk` as well. For more info see: #19211

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

8. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

17. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
